### ISSUE TITLE
Define the scenario interface

### ISSUE THREAD

**Issue by josephburnett** [pivotal/skenario#62]
Currently various traffic patterns are defined as a [golang object that the UI and core simulator both know about](https://github.com/pivotal/skenario/blob/7f37e52d3c7a37ddad7daa1c2bd8bdd0c402f96f/pkg/serve/run_handler.go#L60).  And the same for the [metrics returned to the UI](https://github.com/pivotal/skenario/blob/7f37e52d3c7a37ddad7daa1c2bd8bdd0c402f96f/pkg/serve/run_handler.go#L142).  This should be defined as a protobuf similar to #58 so that the UI can be more decoupled from the core.  This will allow command line / library usage of Skenario core in addition to the UI.

The proto should include an enumerated set of patterns.  But also allow for raw data input (e.g. qps over time).

**Comment by jchesterpivotal** [comment]
Is the idea that traffic patterns will live in a separate process communicating with the simulator core?

**Comment by josephburnett** [comment]
No, I just want to have clearly defined traffic shape which the simulator will implement.  So the UI can be decoupled from the simulator.
